src/Compilers/Core/Portable/Compilation/OperationExtensions.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Semantics
+{
+    internal interface IOperationSearchable
```

Comment:
Perhaps return immediate children only, or express as a visitor (with a VisitChildren() method here), so the caller has some control over the traversal, and so the hierarchy is preserved.